packages/jupyter-ai/jupyter_ai/personas/persona_manager.py

```diff
@@ -260,3 +268,116 @@ def get_chat_dir(self) -> str:
         """
         abspath = self.get_chat_path(absolute=True)
         return os.path.dirname(abspath)
+
+
+class LocalPersonaLoader(LoggingConfigurable):
+    """
+    Load _persona class declarations_ from the local filesystem.
+
+    Those class declarations are then used to instantiate personas by the `PersonaManager`.
+
+    TODO: wire to local .jupyter directory system, for now just use the current
+    directory.
+    """
+
+    def __init__(
+        self,
+        *args,
+        root_dir: str | None = None,
+        **kwargs,
+    ):
+        # Forward other arguments to parent class
+        super().__init__(*args, **kwargs)
+
+        # Set default root directory to current working directory if not provided
+        self.root_dir = root_dir or os.getcwd()
+
+        self.log.info(
+            f"LocalPersonaLoader initialized with root directory: {self.root_dir}"
+        )
+
+    def load_persona_classes(self) -> list[type[BasePersona]]:
+        """
+        Loads persona classes from Python files in the local filesystem.
+
+        Scans the root_dir for .py files, dynamically imports them, and extracts
+        any class declarations that are subclasses of BasePersona.
+        """
+        persona_classes: list[type[BasePersona]] = []
+
+        # Check if root directory exists
+        if not os.path.exists(self.root_dir):
+            self.log.info(f"Root directory does not exist: {self.root_dir}")
+            return persona_classes
+
+        # Find all .py files in the root directory that contain "persona" in the name
+        all_py_files = glob(os.path.join(self.root_dir, "*.py"))
```

Comment:
I think this could trigger a permission error.

Reply:
Wrapped in try/except now.